myosuite/envs/myo/myochallenge/chasetag_v0.py

```diff
@@ -37,11 +37,20 @@ class ChallengeOpponent:
     Contains several different policies. For the final evaluation, an additional
     non-disclosed policy will be used.
     """
-    def __init__(self, sim, rng, probabilities: list, min_spawn_distance: float):
+    def __init__(self,
+                 sim,
+                 rng,
+                 probabilities: tuple[float],
+                 min_spawn_distance: float,
+                 chase_vel_range: tuple[float],
+                 random_vel_range: tuple[float],
+        ):
         self.dt = 0.01
         self.sim = sim
         self.opponent_probabilities = probabilities
         self.min_spawn_distance = min_spawn_distance
+        self.chase_vel_range = chase_vel_range
+        self.random_vel_range = random_vel_range
         self.reset_opponent(rng=rng)
 
     def reset_noise_process(self):
```

Comment:
Do we know if this is deterministic, given seed?

Reply:
yes, the sampling is drawn from the `rng` in the opponent class, which gets set by the outer environment 
- at creation
- every time there is a reset

This ensures that the number generators are always in sync if a seed is set before the episode reset